File: 2022/test/day_18_test.exs
defmodule Day18Test do
  use ExUnit.Case

  import TestHelper
  import Day18

  test "part1" do
    assert part1(input("day_18_example_1")) == 10
    assert part1(input("day_18_example_2")) == 64
    assert part1(input("day_18")) == 456
  end

  test "part2" do
    # assert part2(input("day_18_example_1")) == 123
    # assert part2(input("day_18")) == 456
  end
end

File: 2024/day18/day18_test.go
package day18

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestPart1(t *testing.T) {
	assert.Equal(t, 22, Part1("input_example_1", 6, 12))
	assert.Equal(t, 370, Part1("input", 70, 1024))
}

func TestPart2(t *testing.T) {
	assert.Equal(t, "6,1", Part2("input_example_1", 6, 12))
	assert.Equal(t, "65,6", Part2("input", 70, 1024))
}
